syft/source/location_set.go

```diff
@@ -4,10 +4,13 @@ import (
 	"sort"
 
 	"github.com/mitchellh/hashstructure/v2"
+
+	"github.com/anchore/syft/internal/log"
 )
 
 type LocationSet struct {
-	set map[Location]struct{}
+	set      map[LocationData]struct{}
+	metadata map[LocationData]LocationMetadata
```

Comment:
Why not just modify this to be something like:
```go
type LocationSet struct {
	set map[Location]*Location
```
It seems a bit odd to specialize the metadata here, rather than finding the location from another equal location

Reply:
I condensed it into a single map 👍